nzedb/processing/tv/TVMaze.php

```diff
@@ -0,0 +1,400 @@
+							continue;
+						}
+
+						// Download all episodes if new show to reduce API usage
+						if ($this->countEpsByVideoID($videoId) === false) {
+							$this->getEpisodeInfo($tvmazeid, -1, -1, '', $videoId);
+						}
+
+						// Check if we have the episode for this video ID
+						$episode = $this->getBySeasonEp($videoId, $seasonNo, $episodeNo, $release['airdate']);
+
+						if ($episode === false) {
+							// Send the request for the episode to TVMaze
+							$tvmazeEpisode = $this->getEpisodeInfo(
+									$tvmazeid,
+									$seasonNo,
+									$episodeNo,
+									$release['airdate']
+							);
+
+							if ($tvmazeEpisode) {
+								$episode = $this->addEpisode($videoId, $tvmazeEpisode);
+							}
+						}
+
+						if ($episode !== false && is_numeric($episode) && $episode > 0) {
+							// Mark the releases video and episode IDs
+							$this->setVideoIdFound($videoId, $row['id'], $episode);
+							if ($this->echooutput) {
+								echo $this->pdo->log->primary("Found TVMaze Match!");
+							}
+							continue;
+						}
+					}
+				} //Processing failed, set the episode ID to the next processing group
+				$this->setVideoNotFound(parent::PROCESS_TMDB, $row['id']);
+			}
+		}
+	}
+
+	/**
+	 * Calls the API to lookup the TvMaze info for a given TVDB or TVRage ID
+	 * Returns a formatted array of show data or false if no match
+
+	 * @param $site
+	 * @param $siteId
+	 *
+	 * @return array|bool
+	 */
+	protected function getShowInfoBySiteID($site, $siteId)
+	{
+		$return = $response = false;
+
+		//Try for the best match with AKAs embedded
+		$response = $this->client->getShowBySiteID($site, $siteId);
+
+		sleep(1);
+
+		if (is_array($response)) {
+			$return = $this->formatShowArr($response);
+		}
+		return $return;
+	}
+
+	/**
+	 * Calls the API to perform initial show name match to TVDB title
+	 * Returns a formatted array of show data or false if no match
+	 *
+	 * @param $cleanName
+	 *
+	 * @return array|bool
+	 */
+	protected function getShowInfo($cleanName)
+	{
+		$return = $response = false;
+
+		//Try for the best match with AKAs embedded
+		$response = $this->client->singleSearch($cleanName);
+
+		sleep(1);
+
+		if (is_array($response)) {
+			$return = $this->matchShowInfo($response, $cleanName);
+		}
+		if ($return === false) {
+			//Try for the best match via full search (no AKAs can be returned but the search is better)
+			$response = $this->client->search($cleanName);
+			if (is_array($response)) {
+				$return = $this->matchShowInfo($response, $cleanName);
+			}
+		}
+		//If we didn't get any aliases do a direct alias lookup
+		if (is_array($return) && empty($return['aliases']) && is_numeric($return['tvmaze'])) {
+			$return['aliases'] = $this->client->getShowAKAs($return['tvmaze']);
+		}
+		return $return;
+	}
+
+	/**
+	 * @param $showArr
+	 * @param $cleanName
+	 *
+	 * @return array|bool
+	 */
+	private function matchShowInfo($showArr, $cleanName)
```

Comment:
Alright so why am I doing just array $show here and in other places you've told me to do array $array = [] ?  Please, consistent if you're citing some standard.